examples/DrawXNative/DrawXShared/SwatchColor.cs

```diff
@@ -0,0 +1,81 @@
+﻿////////////////////////////////////////////////////////////////////////////
+//
+// Copyright 2014 Realm Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+////////////////////////////////////////////////////////////////////////////
+
+using System.Collections.Generic;
+using SkiaSharp;
+
+namespace DrawXShared
+{
+    public struct SwatchColor
+    {
+        public string Name;
+        public SKColor Color;
+        private static Dictionary<string, SwatchColor> _colors;
+        private static List<SwatchColor> _indexedColors;
+
+        public static Dictionary<string, SwatchColor> ColorsByName
+        {
+            get
+            {
+                if (_colors == null)
+                {
+                    InitColors();
+                }
+
+                return _colors;
+            }
+        }
+
+        // iterating this array gives you the colours in our desired order
+        public static List<SwatchColor> Colors
```

Comment:
No, it is explicitly treated as a List elsewhere and I don't think the extra effort to do more abstraction is justified.